### PR TITLE
feat: string formatter util functions

### PR DESCRIPTION
**Pull request by kpranoto-aws**
*Changes:*
* Fixed typo for library type
* Added util formatters and tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by SwaySway — APPROVED** [review]
🚢

**Review by Jshhhh — APPROVED** [review]
LGTM

File: packages/codegen-ui/lib/utils/string-formatter.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    case 'Mmm DD, YYYY':
+      return `${validDate.toLocaleString('en-us', { month: 'short' })} ${day}, ${year}`;
+    default:
+      return date;
+  }
+}
+
+export function formatTime(time: string, format: TimeFormat['timeFormat']): string {
+  // AWSTime: hh:mm:ss.sss (24hr format - ISO 8601)
+  const splitTime = time.split(/:|Z/);
```
Comment (cwoolum) [inline packages/codegen-ui/lib/utils/string-formatter.ts:52]: What if time is nullish?

File: packages/codegen-ui/lib/utils/string-formatter.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  limitations under the License.
+ */
+
+import { DateFormat, DateTimeFormat, TimeFormat } from '../types';
+
+const invalidDateStr = 'Invalid Date';
+
+export function formatDate(date: string, format: DateFormat['dateFormat']): string {
+  // AWSDate: YYYY-MM-DD (ISO 8601)
+  const validDate = new Date(Date.parse(date));
```
Comment (cwoolum) [inline packages/codegen-ui/lib/utils/string-formatter.ts:23]: What if date is nullish?

File: packages/codegen-ui/lib/utils/string-formatter.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      return validTime.toLocaleTimeString('en-us');
+    default:
+      return time;
+  }
+}
+
+export function formatDateTime(dateTimeStr: string, format: DateTimeFormat['dateTimeFormat']): string {
+  // AWSTimestamp: millis before or after 1970-01-01-T00:00*Z*
+  // AWSDateTime: hh:mm:ss.sss (24hr format - ISO 8601)
+  const dateTime = /^\d+$/.test(dateTimeStr)
```
Comment (cwoolum) [inline packages/codegen-ui/lib/utils/string-formatter.ts:84]: What if dateTimeStr is nullish?

**Review by cwoolum — COMMENTED** [review]
Need to handle null value scenarios

**Pull request merged**